### ISSUE TITLE
Cypher Directives containing $jwt.sub are not working when called over OGM in Beta 

### ISSUE THREAD

**Issue by eriklueth** [neo4j/graphql#3773]
**Describe the bug**
In the new Beta Iam not able to call a cypher that contains a $jwt.sub over the OGM.

This is a bit related to #3772 because I think before the auth was injected globally 

I have a custom resolver that creates something over the OGM and has a cypher with $jwt in the selection set.
It seems like auth is not global anymore. So when I use the OGM to create something and the selection set has a cypher with $jwt.sub in it, I get @neo4j/graphql:execute Neo4jError: Expected parameter(s): jwt because the executed call to neo4j does not contain the jwt in the params.

I tried adding the jwt to the args or context of the OGM Model but that does not help.

Example types
```
type Event {
  userAttending: Boolean!
    @cypher(
      statement: """
      MATCH (user:User {id: $jwt.sub})-[:ATTENDS]->(this)
      RETURN COUNT(user) <> 0 as result
      """
      columnName: "result"
    )

  organisation: Organisation
    @relationship(type: "ORGANISATION_OF_EVENT", direction: OUT)
}


type Organisation {
  id: ID!
  name: String!
  events: [Event!]! @relationship(type: "ORGANISATION_OF_EVENT", direction: IN)
}




```

Example OGM call
```
  const Organisation = context.ogm.model("Organisation");

  const [organisation] = await Organisation.create({
    args: {
      jwt: context.jwt, //This is sadly also not working to add jwt to the neo4j execution params
    },
    selectionSet: `
    {
      organisations {
      __typename
      id
      name
      events {
        id
        name
        userAttending
      }
    }
  }
    `,
    input: {
      name: args.name,
    },
    context: context,
  })
    .then((resp: any) => {
      console.log(resp);
      return resp.organisations;
    })
    .catch(async (e: any) => console.log(e));
  console.log("Success: Created Organisation " + organisation.id);
```

Logs
```
Params:
{
  "this0_name": "test",
  "resolvedCallbacks": {}
} +22s
  @neo4j/graphql:execute Neo4jError: Expected parameter(s): jwt

    at captureStacktrace (C:\Projekte\Dashpoint\dashpoint-graphql-backend\node_modules\neo4j-driver-core\lib\result.js:611:17)
    at new Result (C:\Projekte\Dashpoint\dashpoint-graphql-backend\node_modules\neo4j-driver-core\lib\result.js:105:23)
    at newCompletedResult (C:\Projekte\Dashpoint\dashpoint-graphql-backend\node_modules\neo4j-driver-core\lib\transaction.js:506:12)
    at Object.run (C:\Projekte\Dashpoint\dashpoint-graphql-backend\node_modules\neo4j-driver-core\lib\transaction.js:338:20)
    at Transaction.run (C:\Projekte\Dashpoint\dashpoint-graphql-backend\node_modules\neo4j-driver-core\lib\transaction.js:179:34)
    at ManagedTransaction.run (C:\Projekte\Dashpoint\dashpoint-graphql-backend\node_modules\neo4j-driver-core\lib\transaction-managed.js:56:21)
    at Executor.transactionRun (C:\Projekte\Dashpoint\dashpoint-graphql-backend\node_modules\@neo4j\graphql\dist\classes\Executor.js:152:28)
    at C:\Projekte\Dashpoint\dashpoint-graphql-backend\node_modules\@neo4j\graphql\dist\classes\Executor.js:138:33
    at TransactionExecutor._safeExecuteTransactionWork (C:\Projekte\Dashpoint\dashpoint-graphql-backend\node_modules\neo4j-driver-core\lib\internal\transaction-executor.js:141:26)
    at TransactionExecutor.<anonymous> (C:\Projekte\Dashpoint\dashpoint-graphql-backend\node_modules\neo4j-driver-core\lib\internal\transaction-executor.js:128:46)
    at step (C:\Projekte\Dashpoint\dashpoint-graphql-backend\node_modules\neo4j-driver-core\lib\internal\transaction-executor.js:52:23)
    at Object.next (C:\Projekte\Dashpoint\dashpoint-graphql-backend\node_modules\neo4j-driver-core\lib\internal\transaction-executor.js:33:53)
    at fulfilled (C:\Projekte\Dashpoint\dashpoint-graphql-backend\node_modules\neo4j-driver-core\lib\internal\transaction-executor.js:24:58)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  constructor: [Function],
  code: 'Neo.ClientError.Statement.ParameterMissing',
  retriable: false
} +2s
Error: Expected parameter(s): jwt
    at Model.create (C:\Projekte\Dashpoint\dashpoint-graphql-backend\node_modules\@neo4j\graphql-ogm\dist\classes\Model.js:127:19)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
```

**Expected behavior**
Global jwt injection into the Neo4J Params or the possibility to add things like the jwt into the execution params when calling the OGM Model function like

```
  const [organisation] = await Organisation.create({
    additionalParams: {
      jwt: context.jwt, // Added to neo4j cypher params
    },
    selectionSet: `
    {
      organisations {
      __typename
      id
      name
      events {
        id
        name
        userAttending
      }
    }
  }
    `,
    input: {
      name: args.name,
    },
    context: context,
  })
```

**Comment by neo4j-team-graphql** [comment]
Many thanks for raising this bug report @eriklueth. :bug: We will now attempt to reproduce the bug based on the steps you have provided.

Please ensure that you've provided the necessary information for a minimal reproduction, including but not limited to:
* Type definitions
* Resolvers
* Query and/or Mutation (or multiple) needed to reproduce

If you have a support agreement with Neo4j, please link this GitHub issue to a new or existing Zendesk ticket.

Thanks again! :pray:

**Comment by neo4j-team-graphql** [comment]
Many thanks for raising this bug report @eriklueth. :bug: We will now attempt to reproduce the bug based on the steps you have provided.

Please ensure that you've provided the necessary information for a minimal reproduction, including but not limited to:
* Type definitions
* Resolvers
* Query and/or Mutation (or multiple) needed to reproduce

If you have a support agreement with Neo4j, please link this GitHub issue to a new or existing Zendesk ticket.

Thanks again! :pray:

**Comment by darrellwarde** [comment]
Thanks for all of these reports, this is another one I'll be taking a look at next week. 🙂

**Comment by eriklueth** [comment]
Thanks, again a great weekend. I saw that you have the cypherParams which can be added to the context. With that, you can inject into the cypher params globally and it seems like you could write $cypherParams.jwt.sub in the cypher itself and add the jwt into the context cypherParams to make everything work. 

But this is not really optimal and the best way would be if jwt is injected directly or you can inject it via a special field in the OGM Model call.

**Comment by eriklueth** [comment]
I tried it with adding jwt inside cypherParams. But same problem here. The cypherParams are also not available if the cypher is called by the OGM.

**Comment by eriklueth** [comment]
Hey @darrellwarde,
for me that context is my general context. So I just tried to add it to the ogm call to see if that solves the jwt issue.

![image](https://github.com/neo4j/graphql/assets/12431981/1ce13136-31b4-4097-965b-07162de81eb4)


The context object has the correct jwt parameter when printing it.

**Comment by vpctorr** [comment]
Using `$someVar` instead of `$cypherParams.someVar` worked for me, since https://github.com/neo4j/graphql/commit/e9bf1e619ee71ead228530a9d46834a655686c6d

Not sure whether the intent was to introduce a breaking change though. If so, it hasn't been mentioned in the migration docs.

**Comment by neo4j-team-graphql** [comment]
We've been able to confirm this bug using the steps to reproduce that you provided - many thanks @eriklueth! :pray: We will now prioritise the bug and address it appropriately.

**Comment by neo4j-team-graphql** [comment]
This bug report has been assigned high priority to fix. If you wish to contribute a fix, please branch from `master` and submit your PR with the base set to `master`. Thanks!

**Comment by darrellwarde** [comment]
This has been fixed in #3830 - `jwt` and `token` shouldn't really be supported in the OGM because the authorization features aren't supported. I have removed these fields from the context type for the OGM.

I believe `cypherParams` to be the "correct" solution here, and there was a bug in their injection into the Cypher query. This has now been fixed, so the solution will be:

```ts
const [organisation] = await Organisation.create({
    selectionSet: `
    {
      organisations {
      __typename
      id
      name
      events {
        id
        name
        userAttending
      }
    }
  }
    `,
    input: {
      name: args.name,
    },
    context: {
        cypherParams: {
            jwt: { sub: "1234" }
        }
    },
  })
```
